### PR TITLE
Miscellaneous warnings

### PR DESCRIPTION
**Pull request by mokibit**
This PR fixes miscellaneous warnings and deletes dead code. 

* [not needed] I have created a file in the `doc/newsfragments` directory *IF* it is a
      user-visible change (and make sure to read the `README.md` in that directory)

### CODE REVIEW

**Comment by shymega** [comment]
This is failing, so we need to review this further.

**Comment by shymega** [comment]
OK, good progress - now we just need to fix Windows builds..

**Comment by mokibit** [comment]
@p12tic tests are fixed.

**Review by p12tic — CHANGES_REQUESTED** [review]
Found just a single small issue.

File: src/test/global/gmock.h [hunk 1]
```diff
@@ -28,5 +28,12 @@
 // gmock includes gtest which has a warning on osx106 (signed/unsigned
 // int compare), so include our special header here first to silence
 // the warning.
+
+
+#pragma GCC diagnostic push
```
[the remark this answers is not in the document]
Comment (shymega) [reply src/test/global/gmock.h]: OK, so this has been deleted from the PR, so I suppose this review could be reviewed, but I'd like to be sure it was intentional - @mokibit, could you confirm? Thank you!

[the remark this answers is not in the document]
Comment (mokibit) [reply src/test/global/gmock.h]: @shymega, yes I fixed this.

**Review by p12tic — APPROVED** [review]

File: src/test/global/gmock.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (shymega) [reply src/test/global/gmock.h]: @mokibit Awesome wanted to check before I could merge, but Povilas beat me to it 😅!

**Pull request merged**